src/main/webapp/app/quiz/edit/drag-and-drop-question/edit-drag-and-drop-question.component.ts

```diff
@@ -839,5 +805,63 @@ export class EditDragAndDropQuestionComponent implements OnInit, OnChanges, Afte
      */
     togglePreview(): void {
         this.showPreview = !this.showPreview;
+        this.prepareForSave();
+    }
+
+    /**
+     * @function changesInMarkdown
+     * @desc detection of  changes in the markdown editor
+     */
+    changesInMarkdown(): void {
+        console.log('markdown');
+        this.questionUpdated.emit();
+        this.changeDetector.detectChanges();
+        this.prepareForSave();
+    }
+
+    /**
+     * @function specialCommandsFound
+     * @desc call the specialCommandFoundSave for each element of the array
+     * @param specialCommands {array} text with the corresponding special command
+     */
+    specialCommandsFound(specialCommands: [string, SpecialCommand][]): void {
+        this.cleanupQuestion();
+        specialCommands.forEach(command => this.specialCommandFoundSave(command[0], command[1]));
+    }
+
+    /**
+     * @function  specialCommandFoundSave
+     * @desc detection of  changes in the markdown editor
```

Comment:
this is not a good documentation. please explain what happens here in more detail!